app/components/UserAttendance/AttendanceModal.tsx

```diff
@@ -48,98 +48,99 @@ const Tab = ({ name, index, activePoolIndex, togglePool }: TabProps) => (
   </Button>
 );
 
-type State = {
-  pools: Pool[];
-  filter: string;
-};
+const AttendanceModal = ({
+  title = 'Status',
+  pools,
+  togglePool,
+  selectedPool,
+  isMeeting,
+}: Props) => {
+  const [amendedPools, setAmendedPools] = useState<Pool[]>([]);
+  const [registrations, setRegistrations] = useState<Registration[]>([]);
+  const [filter, setFilter] = useState<string>('');
 
-class AttendanceModal extends Component<Props, State> {
-  state = {
-    pools: [],
-    filter: '',
-  };
-  static defaultProps = {
-    title: 'Status',
-  };
+  useEffect(() => {
+    generateAmendedPools(pools);
+  }, [pools]);
```

Comment:
You could've cleaned this up and removed the unnecessary hooks at the same time but :shrug:

Reply:
Which hook is unnecessary?

Reply:
The `amendedPools`, `registrations` state hooks and the useEffects. So almost all of them :grimacing:

Reply:
what, but they are needed? Sure it can be calculated higher up, but I don't see how that's cleaner